### PR TITLE
Add dims argument to LogSumExpAtom

### LINKED ISSUE #682: Scalar indexing in `logisticloss`
https://github.com/jump-dev/Convex.jl/blob/2978c32c0fcc3abc1cf10704980367cc23f8e1df/src/atoms/LogSumExpAtom.jl#L48

I guess I missed this one in #614

### PR DESCRIPTION
**Pull request by odow**
Closes #682 
Closes #685

### CODE REVIEW

**Comment by odow** [comment]
So perhaps the simplest thing is just if we add a `; dims` argument to relevant of atoms.

**Review by ericphanson — APPROVED** [review]
Lgtm, nice and simple. I think it would be worth carrying over the tests in src/problem_depot/problems/exp.jl from the other PR

File: src/atoms/LogSumExpAtom.jl [hunk 1]
```diff
@@ -28,22 +37,29 @@ curvature(::LogSumExpAtom) = ConvexVexity()
 function evaluate(x::LogSumExpAtom)
     _x = evaluate(x.children[1])
     max_x = maximum(_x)
-    return max_x + log(sum(exp.(_x .- max_x)))
+    return max_x .+ log.(sum(exp.(_x .- max_x); x.dims))
```
Comment (ericphanson) [inline src/atoms/LogSumExpAtom.jl:40]:
```suggestion
    return output(max_x .+ log.(sum(exp.(_x .- max_x); x.dims)))
```
We may want this so we don’t return vectors for scalars

Comment (odow) [reply src/atoms/LogSumExpAtom.jl:40]: I think this just works. I'd already added a test for the scalar case:
```julia
@test evaluate(logsumexp(x; dims = :)) ≈ 6.456193316018123
```

File: src/problem_depot/problems/exp.jl [hunk 2]
```diff
@@ -121,14 +121,55 @@ end
...
     p = minimize(logsumexp(y), y >= 1; numeric_type = T)
-
     if test
         @test problem_vexity(p) == ConvexVexity()
     end
     handle_problem!(p)
     if test
         @test p.optval ≈ log(exp(1) * 5) atol = atol rtol = rtol
     end
+
```
Comment (odow) [inline src/problem_depot/problems/exp.jl:131]:
```suggestion

```

File: src/atoms/LogSumExpAtom.jl [hunk 3]
```diff
@@ -3,17 +3,26 @@
 # Use of this source code is governed by a BSD-style license that can be found
 # in the LICENSE file or at https://opensource.org/license/bsd-2-clause
 
+"""
+    LogSumExpAtom(x::AbstractExpr, dims::Int = 0)
```
Comment (ericphanson) [inline src/atoms/LogSumExpAtom.jl:7]: should this be `:`? also do we want to canonicalize 0 to `:` or something?

Comment (odow) [reply src/atoms/LogSumExpAtom.jl:7]: Oops yeah. I originally did the `0` think that canonicalized to `:`, then I just made it `Union{Int,Colon}`. Did't update the docstring

Comment (odow) [inline src/atoms/LogSumExpAtom.jl:7]:
```suggestion
    LogSumExpAtom(x::AbstractExpr, dims::Union{Colon,Int} = :)
```

**Review by ericphanson — APPROVED** [review]

**Pull request merged**